ledger/ledger-api-grpc-utils/src/main/scala/com/digitalasset/grpc/GrpcException.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.grpc
+
+import io.grpc.{Metadata, Status, StatusException, StatusRuntimeException}
+
+object GrpcException {
+
+  def unapply(arg: Throwable): Option[(Status, Metadata)] =
```

Comment:
`arg` in these situations should be the LUB of the possible inputs for `Some` returns; this does not prevent it from matching wider selectors, but triggers appropriate errors on some impossible matches.

```suggestion
  def unapply(arg: Exception): Option[(Status, Metadata)] =
```